JobLifecycleMetricsTest#multipleJobsSubmittedAndCompleted

Failed on PR Builder http://jenkins.hazelcast.com/job/Hazelcast-pr-builder/6502/

Stacktrace:
```
java.lang.AssertionError: expected:<2> but was:<3>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at com.hazelcast.jet.core.metrics.JmxMetricsChecker.assertMetricValue(JmxMetricsChecker.java:61)
	at com.hazelcast.jet.core.metrics.JobLifecycleMetricsTest.assertJobStatsOnMember(JobLifecycleMetricsTest.java:190)
	at com.hazelcast.jet.core.metrics.JobLifecycleMetricsTest.assertJobStats(JobLifecycleMetricsTest.java:180)
	at com.hazelcast.jet.core.metrics.JobLifecycleMetricsTest.lambda$multipleJobsSubmittedAndCompleted$1(JobLifecycleMetricsTest.java:95)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1262)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1364)
	at com.hazelcast.jet.core.metrics.JobLifecycleMetricsTest.multipleJobsSubmittedAndCompleted(JobLifecycleMetricsTest.java:95)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

Looks like we call `JobExecutionService#completeExecution` twice when an execution is terminated before it began

relevant logs from the test
```
11:46:04,590 DEBUG |multipleJobsSubmittedAndCompleted| - [JobExecutionService] hz.vibrant_easley.generic-operation.thread-1 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Completed execution of job '0698-7de3-fbc2-0001', execution 0698-7de3-fbc3-0001
11:46:04,591 DEBUG || - [JobExecutionService] ForkJoinPool.commonPool-worker-11 - [127.0.0.1]:5702 [dev] [5.0-SNAPSHOT] Completed execution of job '0698-7de3-fbc2-0001', execution 0698-7de3-fbc3-0001
```
First one is called in `JobExecutionService#terminateExecution0` when `executionContext.terminateExecution(mode)` returns false (it means execution was terminated before it began)
and the second is called in `JobExecutionService#beginExecution0`, specifically in `whenCompleteAsync` block which explains the FJP in the log

--- a/hazelcast/src/main/java/com/hazelcast/jet/impl/JobExecutionService.java
+++ b/hazelcast/src/main/java/com/hazelcast/jet/impl/JobExecutionService.java
@@ -478,17 +478,19 @@ public class JobExecutionService implements DynamicMetricsProvider {
      * Completes and cleans up execution of the given job
      */
     public void completeExecution(@Nonnull ExecutionContext executionContext, Throwable error) {
-        executionContexts.remove(executionContext.executionId());
-        JetDelegatingClassLoader jobClassLoader = jobClassloaderService.getClassLoader(executionContext.jobId());
-        try {
-            doWithClassLoader(jobClassLoader, () -> executionContext.completeExecution(error));
-        } finally {
-            if (!executionContext.isLightJob()) {
-                jobClassloaderService.tryRemoveClassloadersForJob(executionContext.jobId(), EXECUTION);
+        ExecutionContext removed = executionContexts.remove(executionContext.executionId());
+        if (removed != null) {
+            JetDelegatingClassLoader jobClassLoader = jobClassloaderService.getClassLoader(executionContext.jobId());
+            try {
+                doWithClassLoader(jobClassLoader, () -> executionContext.completeExecution(error));
+            } finally {
+                if (!executionContext.isLightJob()) {
+                    jobClassloaderService.tryRemoveClassloadersForJob(executionContext.jobId(), EXECUTION);
+                }
+                executionCompleted.inc();
+                executionContextJobIds.remove(executionContext.jobId());
+                logger.fine("Completed execution of " + executionContext.jobNameAndExecutionId());
             }
-            executionCompleted.inc();
-            executionContextJobIds.remove(executionContext.jobId());
-            logger.fine("Completed execution of " + executionContext.jobNameAndExecutionId());
         }
     }
 

